### PR TITLE
Fixes removal of boolean and empty Json object values from request body

### LINKED ISSUE #3201: Set-MgBetaUserLicense not working
### Describe the bug

The cmdlet:
```
Set-MgBetaUserLicense -UserId $adUser.Id -AddLicenses @{SkuId = $Sku.SkuId} -RemoveLicenses @()
```

gives error:
```
Set-MgBetaUserLicense_AssignExpanded: D:\Repos\PFIBDO-ADM-CloudConfiguration\scripts\aad\Configure-Licenses.ps1:261:25
Line |
 261 |  …             Set-MgBetaUserLicense -UserId $adUser.Id -AddLicenses @(@ …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | One or more parameters of the operation 'assignLicense' are missing from the request payload. The missing parameters are: removeLicenses.  Status: 400        
     | (BadRequest) ErrorCode: Request_BadRequest Date: 2025-02-28T20:04:03  Headers: Cache-Control                 : no-cache Vary                          :       
     | Accept-Encoding Strict-Transport-Security     : max-age=31536000 request-id                    : cffa816c-c9ce-468f-843b-0b38595f1df8 client-request-id       
     | : 32915540-5cf1-4543-a811-3bbb3ba324e5 x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"Switzerland
     | North","Slice":"E","Ring":"3","ScaleUnit":"000","RoleInstance":"ZR1PEPF00000664"}} Link                          :
     | <https://developer.microsoft-tst.com/en-us/graph/changes?$filterby=beta,Users&from=2022-02-01&to=2022-03-01>;rel="deprecation";type="text/html" deprecation   
     | : Thu, 17 Feb 2022 23:59:59 GMT sunset                        : Sat, 17 Feb 2024 23:59:59 GMT x-ms-resource-unit            : 1 Date                          
     | : Fri, 28 Feb 2025 20:04:03 GMT

  Recommendation: See service error codes: https://learn.microsoft.com/graph/errors
```

This code was working in the past.

### Expected behavior

It should be possible to assing just a license without removing one. This has be done in the past by adding an empty array in the RemoveLicenses parameter like documented.

### How to reproduce

Try to assign a license without removing one.

### SDK Version

ModuleType Version    PreRelease Name                                ExportedCommands ---------- -------    ---------- ----                                ---------------- Script     2.26.1                Microsoft.Graph.Authentication      {Add-MgEnvironment, Connect-MgGraph, Disconnect-MgGraph, Get-MgContext…} Script     2.26.1                Microsoft.Graph.Beta.Applications   {Add-MgBetaApplicationKey, Add-MgBetaApplicationPassword, Add-MgBetaServicePrincipalTokenSigni…  Script     2.26.1                Microsoft.Graph.Beta.Groups         {Add-MgBetaGroupFavorite, Add-MgBetaGroupSiteContentModelGetByNameToDrive, Add-MgBetaGroupToLi…  Script     2.26.1                Microsoft.Graph.Beta.Identity.Dire… {Complete-MgBetaDirectoryImpactedResource, Complete-MgBetaDirectoryRecommendation, Complete-Mg…  Script     2.26.1                Microsoft.Graph.Beta.Identity.Gove… {Add-MgBetaAccessReviewDecision, Add-MgBetaAccessReviewInstanceDecision, Add-MgBetaIdentityGov…  Script     2.26.1                Microsoft.Graph.Beta.Identity.Sign… {Confirm-MgBetaInformationProtectionSignature, Confirm-MgBetaRiskyServicePrincipalCompromised,…  Script     2.26.1                Microsoft.Graph.Beta.Users          {Complete-MgBetaUserOutlookTask, Complete-MgBetaUserOutlookTaskFolderTask, Complete-MgBetaUser…  Script     2.26.1                Microsoft.Graph.Beta.Users.Actions  {Clear-MgBetaUserAndBlockManagedApp, Clear-MgBetaUserManagedAppRegistrationByAzureAdDeviceId, … 

### Latest version known to work for scenario above?

2.25.X

### Known Workarounds

Downgrading module could work may does not have to

### Debug output

<details><summary>Click to expand log</summary>
```

DEBUG: [CmdletBeginProcessing]: - Set-MgBetaUserLicense begin processing with parameterSet 'AssignExpanded'.
DEBUG: [Authentication]: - AuthType: 'Delegated', TokenCredentialType: 'InteractiveBrowser', ContextScope: 'CurrentUser', AppName: 'Microsoft Graph Command Line Tools'.
DEBUG: [Authentication]: - Scopes: [Application.ReadWrite.All, AppRoleAssignment.ReadWrite.All, AuditLog.Read.All, ChannelMessage.Send, Contacts.Read, CrossTenantInformation.ReadBasic.All, DelegatedPermissionGrant.ReadWrite.All, DeviceManagementApps.Read.All, DeviceManagementApps.ReadWrite.All, DeviceManagementConfiguration.Read.All, DeviceManagementConfiguration.ReadWrite.All, DeviceManagementManagedDevices.Read.All, DeviceManagementManagedDevices.ReadWrite.All, DeviceManagementRBAC.Read.All, DeviceManagementServiceConfig.Read.All, DeviceManagementServiceConfig.ReadWrite.All, Directory.AccessAsUser.All, Directory.Read.All, Directory.ReadWrite.All, Domain.ReadWrite.All, email, Group.ReadWrite.All, GroupMember.ReadWrite.All, openid, Organization.ReadWrite.All, OrganizationalBranding.ReadWrite.All, Policy.Read.All, Policy.ReadWrite.AuthenticationMethod, Policy.ReadWrite.Authorization, Policy.ReadWrite.ConditionalAccess, Policy.ReadWrite.CrossTenantAccess, Policy.ReadWrite.DeviceConfiguration, Policy.ReadWrite.PermissionGrant, profile, RoleAssignmentSchedule.ReadWrite.Directory, RoleEligibilitySchedule.Read.Directory, RoleEligibilitySchedule.ReadWrite.Directory, RoleManagement.Read.All, RoleManagement.ReadWrite.Directory, SharePointTenantSettings.ReadWrite.All, TeamMember.ReadWrite.All, TeamsApp.ReadWrite.All, TeamsAppInstallation.ReadWriteForTeam, TeamsAppInstallation.ReadWriteSelfForTeam, TeamSettings.ReadWrite.All, TeamsTab.ReadWrite.All, User.Read.All, User.ReadWrite.All, UserAuthenticationMethod.Read.All, UserAuthenticationMethod.ReadWrite.All, WindowsUpdates.ReadWrite.All].

Confirm
Are you sure you want to perform this action?
Performing the operation "Set-MgBetaUserLicense_AssignExpanded" on target "Call remote 'POST /users/{user-id}/microsoft.graph.assignLicense' operation".
[Y] Yes  [A] Yes to All  [N] No  [L] No to All  [S] Suspend  [?] Help (default is "Y"): A
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
POST

Absolute Uri:
https://graph.microsoft.com/beta/users/0e88beff-7c45-4bc2-9631-bca7e7e0908b/microsoft.graph.assignLicense

Headers:
FeatureFlag                   : 00000003
Cache-Control                 : no-store, no-cache
User-Agent                    : Mozilla/5.0,(Windows NT 10.0; Microsoft Windows 10.0.26100; de-CH),PowerShell/2025.0.0
SdkVersion                    : graph-powershell-beta/2.26.1
client-request-id             : fb641a05-b41d-4fb6-ab26-24d6bb4974d9
Accept-Encoding               : gzip,deflate,br

Body:
{
  "addLicenses": [
    {
      "skuId": "f245ecc8-75af-4f8e-b61f-27d8114de5f3"
    }
  ]
}


DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Cache-Control                 : no-cache
Vary                          : Accept-Encoding
Strict-Transport-Security     : max-age=31536000
request-id                    : 974af0e7-9da6-49f6-83e8-c257c2b9a881
client-request-id             : fb641a05-b41d-4fb6-ab26-24d6bb4974d9
x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"Switzerland North","Slice":"E","Ring":"3","ScaleUnit":"000","RoleInstance":"ZR1PEPF0000066C"}}
Link                          : <https://developer.microsoft-tst.com/en-us/graph/changes?$filterby=beta,Users&from=2022-02-01&to=2022-03-01>;rel="deprecation";type="text/html"
deprecation                   : Thu, 17 Feb 2022 23:59:59 GMT
sunset                        : Sat, 17 Feb 2024 23:59:59 GMT
x-ms-resource-unit            : 1
Date                          : Fri, 28 Feb 2025 20:06:39 GMT

Body:
{
  "error": {
    "code": "Request_BadRequest",
    "message": "One or more parameters of the operation 'assignLicense' are missing from the request payload. The missing parameters are: removeLicenses.",
    "innerError": {
      "date": "2025-02-28T20:06:39",
      "request-id": "974af0e7-9da6-49f6-83e8-c257c2b9a881",
      "client-request-id": "fb641a05-b41d-4fb6-ab26-24d6bb4974d9"
    }
  }
}


Set-MgBetaUserLicense_AssignExpanded: D:\Repos\PFIBDO-ADM-CloudConfiguration\scripts\aad\Configure-Licenses.ps1:261:25
Line |
 261 |  …             Set-MgBetaUserLicense -UserId $adUser.Id -AddLicenses @(@ …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | One or more parameters of the operation 'assignLicense' are missing from the request payload. The missing parameters are: removeLicenses.  Status: 400        
     | (BadRequest) ErrorCode: Request_BadRequest Date: 2025-02-28T20:06:39  Headers: Cache-Control                 : no-cache Vary                          :       
     | Accept-Encoding Strict-Transport-Security     : max-age=31536000 request-id                    : 974af0e7-9da6-49f6-83e8-c257c2b9a881 client-request-id       
     | : fb641a05-b41d-4fb6-ab26-24d6bb4974d9 x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"Switzerland
     | North","Slice":"E","Ring":"3","ScaleUnit":"000","RoleInstance":"ZR1PEPF0000066C"}} Link                          :
     | <https://developer.microsoft-tst.com/en-us/graph/changes?$filterby=beta,Users&from=2022-02-01&to=2022-03-01>;rel="deprecation";type="text/html" deprecation   
     | : Thu, 17 Feb 2022 23:59:59 GMT sunset                        : Sat, 17 Feb 2024 23:59:59 GMT x-ms-resource-unit            : 1 Date                          
     | : Fri, 28 Feb 2025 20:06:39 GMT

  Recommendation: See service error codes: https://learn.microsoft.com/graph/errors

```
</details>

### Configuration

PS D:\Repos\PFIBDO-ADM-CloudConfiguration> $PSVersionTable 

Name                           Value
----                           -----
PSVersion                      7.5.0
PSEdition                      Core
GitCommitId                    7.5.0
OS                             Microsoft Windows 10.0.26100
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

Architecture: x64

### Other information

_No response_

### LINKED ISSUE #3212: New-MgGroup not working using BodyParameter (asking for mailEnabled value despite being set)
### Describe the bug

I am trying to create a new Microsoft 365 Group (before converting it to a Teams team).

I am trying to create the group using the _New-MgGroup_ Cmdlet along with the _-BodyParameter_ parameter, especially to select the group owner.

The same code used to work in version 2.25 and below, suddenly stopped working in 2.26.1. I installed v2.25 back and it works again with this version.

### Expected behavior

```
PS> New-MgGroup  -BodyParameter $params

DisplayName            Id                                   MailNickname              Description                                                                            GroupTypes
-----------            --                                   ------------              -----------                                                                            ----------
TEST ORG 123 #groupId# test-org-123 test {Unified}
```

### How to reproduce

```
$params = @{
    DisplayName = "TEST ORG 123"
    MailEnabled = $False
    MailNickName = "test-org-123" 
    SecurityEnabled  = $True
    GroupTypes = "Unified" 
    Description = "Test"
}
New-MgGroup -BodyParameter $params
```

Throws the below error : 

```
New-MgGroup : A value is required for property 'mailEnabled' of resource 'Group'.
Status: 400 (BadRequest)
ErrorCode: Request_BadRequest
Date: 2025-03-06T15:52:06
Transfer-Encoding             : chunked
Vary                          : Accept-Encoding
Strict-Transport-Security     : max-age=31536000
request-id                    : 27bd77e5-6831-4dcc-ad57-25134d85e6d3
client-request-id             : 32594ca4-9a5f-48db-9641-e65161ff1234
x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"France Central","Slice":"E","Ring":"5","ScaleUnit":"005","RoleInstance":"PA3PEPF0000F729"}}
x-ms-resource-unit            : 1
Cache-Control                 : no-cache
Date                          : Thu, 06 Mar 2025 15:52:05 GMT
Au caractère Ligne:1 : 1
+ $group = New-MgGroup -BodyParameter $params
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : ({ Headers = , b...oftGraphGroup }:<>f__AnonymousType2`2) [New-MgGroup_Create], Exception
    + FullyQualifiedErrorId : Request_BadRequest,Microsoft.Graph.PowerShell.Cmdlets.NewMgGroup_Create
```

### SDK Version

2.26.1

### Latest version known to work for scenario above?

2.25

### Known Workarounds

It works this way :
`New-MgGroup @params`

But then I can't use the following parameter in the set : 
```
"owners@odata.bind" = @(
    "https://graph.microsoft.com/v1.0/users('#userId#')"
)
```

### Debug output

<details><summary>Click to expand log</summary>

```
DÉBOGUER : [CmdletBeginProcessing]: - New-MgGroup begin processing with parameterSet 'Create'.

DÉBOGUER : [Authentication]: - AuthType: 'AppOnly', TokenCredentialType: 'ClientCertificate', ContextScope: 'Process', AppName: 'CEA Teams creation'.

DÉBOGUER : [Authentication]: - Scopes: [TeamSettings.ReadWrite.All, User.ReadBasic.All, Teamwork.Migrate.All, Sites.ReadWrite.All, Team.Create, Group.ReadWrite.All].

DÉBOGUER : ============================ HTTP REQUEST ============================

HTTP Method:
POST

Absolute Uri:
https://graph.microsoft.com/v1.0/groups

Headers:
FeatureFlag                   : 00000003
Cache-Control                 : no-store, no-cache
User-Agent                    : Mozilla/5.0,(Windows NT 10.0; Microsoft Windows 10.0.19045; fr-FR),PowerShell/5.1.19041.5486
SdkVersion                    : graph-powershell/2.26.1
client-request-id             : 65fa1cb0-5fbb-4ffa-a03b-55edfdca6c49

Body:
{
  "description": "Test",
  "displayName": "TEST ORG 123",
  "groupTypes": [
    "Unified"
  ],
  "mailNickname": "test-org-123"
}

DÉBOGUER : ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Transfer-Encoding             : chunked
Vary                          : Accept-Encoding
Strict-Transport-Security     : max-age=31536000
request-id                    : f30ba99b-a74a-4ed1-8bd9-b84f40e3a541
client-request-id             : 65fa1cb0-5fbb-4ffa-a03b-55edfdca6c49
x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"France Central","Slice":"E","Ring":"5","ScaleUnit":"001","RoleInstance":"PA2PEPF00004EB9"}}
x-ms-resource-unit            : 1
Cache-Control                 : no-cache
Date                          : Thu, 06 Mar 2025 16:02:32 GMT

Body:
{
  "error": {
    "code": "Request_BadRequest",
    "message": "A value is required for property 'mailEnabled' of resource 'Group'.",
    "innerError": {
      "date": "2025-03-06T16:02:33",
      "request-id": "f30ba99b-a74a-4ed1-8bd9-b84f40e3a541",
      "client-request-id": "65fa1cb0-5fbb-4ffa-a03b-55edfdca6c49"
    }
  }
}


A value is required for property 'mailEnabled' of resource 'Group'.

Status: 400 (BadRequest)
ErrorCode: Request_BadRequest
Date: 2025-03-06T16:02:33

Headers:
Transfer-Encoding             : chunked
Vary                          : Accept-Encoding
Strict-Transport-Security     : max-age=31536000
request-id                    : f30ba99b-a74a-4ed1-8bd9-b84f40e3a541
client-request-id             : 65fa1cb0-5fbb-4ffa-a03b-55edfdca6c49
x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"France Central","Slice":"E","Ring":"5","ScaleUnit":"001","RoleInstance":"PA2PEPF00004EB9"}}
x-ms-resource-unit            : 1
Cache-Control                 : no-cache
Date                          : Thu, 06 Mar 2025 16:02:32 GMT


New-MgGroup : A value is required for property 'mailEnabled' of resource 'Group'.
Status: 400 (BadRequest)
ErrorCode: Request_BadRequest
Date: 2025-03-06T16:02:33
Headers:
Transfer-Encoding             : chunked
Vary                          : Accept-Encoding
Strict-Transport-Security     : max-age=31536000
request-id                    : f30ba99b-a74a-4ed1-8bd9-b84f40e3a541
client-request-id             : 65fa1cb0-5fbb-4ffa-a03b-55edfdca6c49
x-ms-ags-diagnostic           : {"ServerInfo":{"DataCenter":"France Central","Slice":"E","Ring":"5","ScaleUnit":"001","RoleInstance":"PA2PEPF00004EB9"}}
x-ms-resource-unit            : 1
Cache-Control                 : no-cache
Date                          : Thu, 06 Mar 2025 16:02:32 GMT
Au caractère Ligne:1 : 1
+ New-MgGroup -BodyParameter $params -Debug
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidOperation : ({ Headers = , b...oftGraphGroup }:<>f__AnonymousType2`2) [New-MgGroup_Create], Exception
    + FullyQualifiedErrorId : Request_BadRequest,Microsoft.Graph.PowerShell.Cmdlets.NewMgGroup_Create
DÉBOGUER : [CmdletEndProcessing]: - New-MgGroup end processing.
```

</details>

### Configuration

Powershell integrated terminal in VS Code
```
PS> $psversiontable

Name                           Value
----                           -----
PSVersion                      5.1.19041.5486
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.19041.5486
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

### Other information

The request body seems truncated in the HTTP post request.

### PR DESCRIPTION
**Pull request by timayabi2020**
Fixes #3201, #3212 #3215, #3202, #2902

The PR also updates tests to cover both boolean and empty Json object valued property scenarios.

**_Ensuring boolean property values are not dropped_**
 
image

**_Ensures empty Json object/list as a property value is not dropped_**

**Example 1:**
image

**Example 2:**
image

### CODE REVIEW

File: tools/Tests/JsonUtilitiesTest/JsonExtensionsTests.cs [hunk 1]
```diff
@@ -146,69 +148,42 @@ public void RemoveDefaultNullProperties_ShouldNotAlterValidData()
...
-        // Act
-        string result = input.ReplaceAndRemoveSlashes();
+        string result = inputJson.RemoveDefaultNullProperties();
 
         // Assert
-        Assert.Equal(expected, result);
-    }
-
-    [Fact]
-    public void ReplaceAndRemoveSlashes_Should_Handle_Invalid_Json_Gracefully()
```
Comment (andrueastman) [inline tools/Tests/JsonUtilitiesTest/JsonExtensionsTests.cs:202]: Just to confirm here, is the removing of these tests intentional?

**Review by andrueastman — APPROVED** [review]

**Comment by timtroch** [comment]
This doesn't seem to be resolved yet:

image

**Comment by Xuzhou-Huang** [comment]
Will this fix get released in next version?

**Comment by zaaj** [comment]
@timtroch ,

Looks like it takes more than an hour for Microsoft to go from accepting a bug-fix pull-request to releasing a new version of the Microsoft.Graph powershell module into the Powershell Gallery.

@Xuzhou-Huang, I'm not a Microsoft employee, but since the pull request got accepted a few hours ago, I would expect this bug fix will make it's way into a future release - I don't know their release cadence so can't say when or in what version these fixes will be included.  

In the mean time, uninstalling the 2.26.1 release and re-installing 2.25.0 should help.

**Comment by jolscr** [comment]
related to issue #3212 
Issue not fixed with release 2.27.0 Microsoft.Graph.
image

Some parameters removed from query ??

image

image
